supertokens/supertokens-core
# Added Tests

## Related Issues
- #80 

## Changes
- Tests added:
   - EmailPassword tests:
      - Check that StorageLayer.getEmailPasswordStorageLayer throws an exception if the storage type is not SQL (and
       vice versa)
      - Test normaliseEmail function
      - Test that the reset password token length is 128 and has URL safe characters (generate a token 100 times and
      for each, check the above).
      - After sign up, check that the password is hashed in the db
      - After reset password generate token, check that the token is hased in the db
      - After reset password completed, check that the password is hashed in the db
   - GetUserAPI test
     - Check for bad input (missing fields)
     - Check good input works
     - Check for all types of output 
   - GeneratePasswordResetToken test
     - Check for bad input (missing fields)
     - Check good input works
     - Check for all types of output
   - ResetPasswordAPI tests
     - Check for bad input (missing fields)
     - Check good input works
     - Check for all types of output
     - Check that invalid method throws 400 error
     - Check that empty password throws 400 error
   - SignInAPI tests
     - Check for bad input (missing fields)
     - Check good input works
     - Test that sign in with unnormalised email like Test@gmail.com should also work
     - Test that giving an empty password, empty email, invalid email, random email or wrong password throws a wrong
       credentials error
   - SignUpAPI tests
     -  Check for bad input (missing fields)
     - Check good input works and that user is there in db (and then call sign in)
     - Test the normalise email function
     - Test that only the normalised email is saved in the db
     - Test that giving an empty password throws a bad input error

## Migration:
  - none
